File: packages/urdu-js/src/normalization/alif.ts
import { ALIF_VARIANTS } from './unicode-data.js'

const ALIF_RE = new RegExp(
  '[' + ALIF_VARIANTS.map(cp => String.fromCodePoint(cp)).join('') + ']',
  'g'
)

export function normalizeAlif(text: string): string {
  return text.replace(ALIF_RE, '\u0627')
}

File: packages/urdu-dart/lib/src/normalization/alif.dart
import 'unicode_data.dart';

final RegExp _alifRe = RegExp(
  '[${alifVariants.map((cp) => String.fromCharCode(cp)).join()}]',
);

/// Replaces all alif variants with bare alif (U+0627).
String normalizeAlif(String text) => text.replaceAll(_alifRe, '\u0627');
